src/nvim/api/vim.c

```diff
@@ -2450,3 +2451,185 @@ void nvim_del_user_command(String name, Error *err)
+    if (len > 0) {
+      ADD(args, STRING_OBJ(cstrn_to_string(buf, len)));
+      i++;
+    }
+  }
+
+  if (ea.cmdidx == CMD_USER) {
+    PUT(result, "cmd", CSTR_TO_OBJ((char *)USER_CMD(ea.useridx)->uc_name));
+  } else if (ea.cmdidx == CMD_USER_BUF) {
+    PUT(result, "cmd", CSTR_TO_OBJ((char *)USER_CMD_GA(&curbuf->b_ucmds, ea.useridx)->uc_name));
+  } else {
+    PUT(result, "cmd", CSTR_TO_OBJ((char *)get_command_name(NULL, ea.cmdidx)));
+  }
+  PUT(result, "line1", INTEGER_OBJ(ea.line1));
+  PUT(result, "line2", INTEGER_OBJ(ea.line2));
+  PUT(result, "force", BOOLEAN_OBJ(ea.forceit));
+  PUT(result, "args", ARRAY_OBJ(args));
+
+  if (ea.argt & EX_EXTRA) {
+    if (ea.argt & EX_NOSPC) {
+      if (ea.argt & EX_NEEDARG) {
+        nargs[0] = '1';
+      } else {
+        nargs[0] = '?';
+      }
+    } else if (ea.argt & EX_NEEDARG) {
+      nargs[0] = '+';
+    } else {
+      nargs[0] = '*';
+    }
+  } else {
+    nargs[0] = '0';
+  }
+  nargs[1] = '\0';
+  PUT(result, "nargs", CSTR_TO_OBJ(nargs));
+
+  switch (ea.addr_type) {
+  case ADDR_LINES:
+    STRCPY(addr, "line");
+    break;
+  case ADDR_ARGUMENTS:
+    STRCPY(addr, "arg");
+    break;
+  case ADDR_BUFFERS:
+    STRCPY(addr, "buf");
+    break;
+  case ADDR_LOADED_BUFFERS:
+    STRCPY(addr, "load");
+    break;
+  case ADDR_WINDOWS:
+    STRCPY(addr, "win");
+    break;
+  case ADDR_TABS:
+    STRCPY(addr, "tab");
+    break;
+  case ADDR_QUICKFIX:
+    STRCPY(addr, "qf");
+    break;
+  case ADDR_NONE:
+    STRCPY(addr, "none");
+    break;
+  default:
+    STRCPY(addr, "?");
+    break;
+  }
+  PUT(result, "addr", CSTR_TO_OBJ(addr));
+  PUT(result, "nextcmd", CSTR_TO_OBJ((char *)ea.nextcmd));
+
+  Dictionary mods = ARRAY_DICT_INIT;
+  PUT(mods, "silent", BOOLEAN_OBJ(cmdinfo.silent));
+  PUT(mods, "emsg_silent", BOOLEAN_OBJ(cmdinfo.emsg_silent));
+  PUT(mods, "sandbox", BOOLEAN_OBJ(cmdinfo.sandbox));
+  PUT(mods, "noautocmd", BOOLEAN_OBJ(cmdinfo.noautocmd));
+  PUT(mods, "tab", INTEGER_OBJ(cmdmod.tab));
+  PUT(mods, "verbose", INTEGER_OBJ(cmdinfo.verbose));
+  PUT(mods, "browse", BOOLEAN_OBJ(cmdmod.browse));
+  PUT(mods, "confirm", BOOLEAN_OBJ(cmdmod.confirm));
+  PUT(mods, "hide", BOOLEAN_OBJ(cmdmod.hide));
+  PUT(mods, "keepalt", BOOLEAN_OBJ(cmdmod.keepalt));
+  PUT(mods, "keepjumps", BOOLEAN_OBJ(cmdmod.keepjumps));
+  PUT(mods, "keepmarks", BOOLEAN_OBJ(cmdmod.keepmarks));
+  PUT(mods, "keeppatterns", BOOLEAN_OBJ(cmdmod.keeppatterns));
+  PUT(mods, "lockmarks", BOOLEAN_OBJ(cmdmod.lockmarks));
+  PUT(mods, "noswapfile", BOOLEAN_OBJ(cmdmod.noswapfile));
+
+  Dictionary splitmods = ARRAY_DICT_INIT;
+  PUT(splitmods, "vertical", BOOLEAN_OBJ(cmdmod.split & WSP_VERT));
+  PUT(splitmods, "aboveleft", BOOLEAN_OBJ(cmdmod.split & WSP_ABOVE));
+  PUT(splitmods, "belowright", BOOLEAN_OBJ(cmdmod.split & WSP_BELOW));
+  PUT(splitmods, "topleft", BOOLEAN_OBJ(cmdmod.split & WSP_TOP));
+  PUT(splitmods, "botright", BOOLEAN_OBJ(cmdmod.split & WSP_BOT));
```

Comment:
I would naively think these are mutually exclusive, so perhaps we could instead have `split = "aboveleft"" attribute instead of four invididual ones? (still separate flag for vertical tho)

Reply:
> I would naively think these are mutually exclusive, so perhaps we could instead have `split = "aboveleft"" attribute instead of four invididual ones? (still separate flag for vertical tho)

Well, I'd have done it too but it's a bit more complicated than that. For example `aboveleft = false` doesn't imply that the `belowright` modifier is set, and vice-versa.

Reply:
> For example aboveleft = false doesn't imply that the belowright modifier is set, and vice-versa

That's not necessary. the value could empty as well.

>  Also aboveleft and topleft can both be true at the same time afaik, same with belowright and botright

Is the behaviour with topleft+aboveleft different than just topleft? Sounds like `topleft` would just take priority then.

Reply:
Alright, fixed it